src/homework6/police.py

```diff
@@ -0,0 +1,131 @@
+"""Create simulator from real life. This can be booking room in hotel,
+visit to casino, visit to bar. Create 3-4 objects, that can describe
+situation. Objects should contain attributes and methods to simulate some
+use cases. Completed program should print object states, it actions (
+methods) and objects interaction."""
+# flake8: noqa:H238
+import random
+import sys
+
+
+class FightUnit:
+    def __init__(self, strength=random.randint(1, 20)):
+        self.strength = strength
+
+    def throw_dice(self):
+        return random.randint(1, self.strength + 1)
+
+
+class CrimeFighter(FightUnit):
+    strength = 0
+
+    def fight_gang(self, gang):
+        self_luck, gang_luck = self.throw_dice(), gang.throw_dice()
+        if (self_luck > gang_luck):
+            self.strength -= 1
+            gang.is_eliminated()
+        elif (self_luck == gang_luck):
+            self.strength = 0
+            gang.is_eliminated()
+        else:
+            self.strength = 0
+
+
+class PoliceSquad(CrimeFighter):
+    def fight_gang(self, gang):
+        super().fight_gang(gang)
+        if not self.strength:
+            print("Police squad is destroyed")
+
+
+class Gang(FightUnit):
+    def is_eliminated(self):
+        self.strength = 0
+        print("Gang was eliminated!")
+
+
+class OrganizedStructure:
+    group_name = ""
+    group_type = FightUnit
```

Comment:
I don't understand why we need these vars here as class variable (remember that instance of this class would be Organized Structure and we can put them to instance variables.